### ISSUE TITLE
Remote regression from 1.8.4 to 1.9.0, multiple remotes hit downloading a package

### ISSUE THREAD

**Issue by brianembry** [conan-io/conan#3882]
To help us debug your issue please explain:

- [* ] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [* ] I've specified the Conan version, operating system version and any tool that can be relevant.
- [*] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

As before, I have a particular centos5 slave that doesn't have an upgraded ssl and really doesn't like it when conan tries to talk to bintray (crash). Machine is configured with an artifactory remote first, bintray second, but bintray remote should never be hit as all packages are in artifactory.

$ conan remote list
conan-local: https://artifactory.<removed>/api/conan/conan-local [Verify SSL: True]
conan-center: https://conan.bintray.com [Verify SSL: True]

We then `$ conan install . --build=missing` 

Now in below example libmagic is the first full hit where binaries are available for this system ("Download"). gtest is package only (being built). 

i.e.
```
13:23:49 [linux_i686]     bzip2/1.0.6@jenkins/stable:1951ee17b0cc43d5963ddb6e1174db19bcc85137 - Build
13:23:49 [linux_i686]     gtest/1.8.0@jenkins/stable:caef3bc369421e559b530d159bebf04d0448c076 - Build
13:23:49 [linux_i686]     libmagic/5.33@jenkins/stable:2add8c254ae251eda8b741b469172d309bd42b7d - Download
```
Anyway, snip of the logs
```
[linux_x86_64] gtest/1.8.0@jenkins/stable: Not found in local cache, looking in remotes...
[linux_x86_64] gtest/1.8.0@jenkins/stable: Trying with 'conan-local'...
[linux_x86_64] Downloading conanmanifest.txt
[linux_x86_64]
[linux_x86_64] Downloading conanfile.py
[linux_x86_64]
[linux_x86_64] Downloading conan_export.tgz
[linux_x86_64]
[linux_x86_64] ....
[linux_x86_64] libmagic/5.33@jenkins/stable: Not found in local cache, looking in remotes...
[linux_x86_64] libmagic/5.33@jenkins/stable: Trying with 'conan-local'...
[linux_x86_64] Downloading conanmanifest.txt
[linux_x86_64]
[linux_x86_64] Downloading conanfile.py
[linux_x86_64]
[linux_x86_64] ERROR: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
[linux_x86_64]
[linux_x86_64] Unable to connect to conan-center=https://conan.bintray.com
[linux_x86_64] /usr/local/lib/python3.5/site-packages/urllib3/util/ssl_.py:352: SNIMissingWarning: An HTTPS request has been made, but the SNI (Server Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
[linux_x86_64]   SNIMissingWarning
```

So, I believe in the case of multiple remotes, with conan 1.9 both remotes are contacted when a package is downloaded (but not when a package is built). The build slave is happy if I pin conan to 1.8.4.

**Comment by lasote** [comment]
It looks like it is trying to search for the binary in the second remote without following the remote used to download the recipe. I will try to confirm it tomorrow.

As it is a different behaviour from 1.8 I assume that it should be fixed, but this flow have some benefits like having different binaries (maybe for different architectures) in a second repository, for example I have a personal repository with ARM packages for some Conan-center packages. A simple install will resolve all the binaries.

In your case, shouldn't be better for you to specify the remote with "-r" or removing Conan center remote if it shouldn't be contacted?

**Comment by brianembry** [comment]
I agree I can remove conan-center, and probably will. Figured I would raise it just in case it was unexpected.

I didn't think of the ARM usecase you mentioned above, how do you ensure packages are in sync across multiple repo's? How do you resolve differences between the conanfiles/exports etc. between the repos (assuming same version)?

**Comment by lasote** [comment]
Very good point, the answer is in the package revisions #3055 feature. Still a work in progress but Conan will be progressively introducing that feature until Conan 2.0 is released.

With package revisions a recipe is unique and you could have different binaries for the same recipe revision in different remotes. You could safety mix them because you can guarantee that the used recipe to generate the binaries is the same.

So when revisions feature is fully deployed the mixing of binaries from different remotes will be a valid flow.

Until then, I agree we need to revert the behaviour to the previous one (1.8.4)

**Comment by lasote** [comment]
I tried to reproduce the issue without success. Are you using `config install` to install a `registry.txt` file maybe?

**Comment by lasote** [comment]
Could you send me the contents of the `registry.json` file in your slave? It could be that for some reason, I still don't know, in the registry file the reference for the `libmagic` is pointing to the bintray remote.

**Comment by lasote** [comment]
Hi, @memsharded has found a bug. It should be solved in 1.9.1
